### PR TITLE
Redesign password recovery email

### LINKED ISSUE #1066: Redesign Password Recovery email
When a user requests a password recovery, the email they receive leaves a bit to be desired as far as the user experience is concerned.

I propose to redesign the email a bit, with a better email subject line and better content body.

Here is a screenshot of what it currently looks like (sensitive data blurred):
recovery

The following items can be improved:

**Subject line**
https://github.com/revive-adserver/revive-adserver/blob/master/lib/max/language/en/default.lang.php#L1078 is the string used for the mail subject. However, the "%s" in "%s password recovery" does not get replaced with anything in https://github.com/revive-adserver/revive-adserver/blob/master/lib/OA/Admin/PasswordRecovery.php#L247 . I assume that %s is supposed to be application name from $aPref['name'].

**Content**
The body of the email could do with a bit more text to explain what it is, why it was sent, and what the recipient is supposed to do. Also, it should have some text to help a recipient who did not request a password recovery (since that could indicate an ongoing attempt at compromising their password).

A opening greeting ("Dear %name,") and a signature should also be part of the content.

### PR DESCRIPTION
**Pull request by andrewatfornax**
Fix #1066

### CODE REVIEW

File: lib/OA/Admin/PasswordRecovery.php [hunk 1]
```diff
@@ -217,37 +217,66 @@ function checkAccess()
...
+            // Load the appropriate language details for the email recipient
+            Language_Loader::load('default', $u['language']);
+
+            // Generate the body of the password reset email for this user
+            $emailBody = $GLOBALS['strPwdRecEmailBody'];
+            $emailBody = str_replace('{name}', $u['contact_name'], $emailBody);
+            $emailBody = str_replace('{username}', $u['username'], $emailBody);
+            $emailBody = str_replace('{reset_link}', $recoveryUrl, $emailBody);
+            if (!empty($aConf['email']['fromName']) && !empty($aConf['email']['fromAddress'])) {
+                $adminSignature = "Sincerely,\n\n{$aConf['email']['fromName']}\n{$aConf['email']['fromAddress']}";
```
Comment (mbeccati) [inline lib/OA/Admin/PasswordRecovery.php:261]: Signatures should probably be translatable too.

File: lib/OA/Admin/PasswordRecovery.php [hunk 2]
```diff
@@ -217,37 +217,66 @@ function checkAccess()
...
+            }
+            if (!empty($aConf['email']['fromAddress'])) {
+                // Use the administrative account details
+                $emailFromAddress = $aConf['email']['fromAddress'];
+            } else {
+                $emailFromAddress = $email;
             }
 
-            // Hack
-            $aConf['email']['admin_name'] = $aPref['admin_fullname'];
```
Comment (mbeccati) [inline lib/OA/Admin/PasswordRecovery.php:243]: What about this "hack"? I suppose it was there for a reason?

File: lib/max/language/de/default.lang.php [hunk 3]
```diff
@@ -895,7 +895,6 @@
 $GLOBALS['strPwdRecWrongId'] = "Falsche ID";
 $GLOBALS['strPwdRecEnterEmail'] = "Geben Sie nachfolgend Ihre eMail Adresse ein";
 $GLOBALS['strPwdRecEnterPassword'] = "Geben Sie nachfolgend Ihr neues Passwort ein";
-$GLOBALS['strPwdRecResetLink'] = "Link zum Passwort zurücksetzen";
```
Comment (mbeccati) [inline lib/max/language/de/default.lang.php:898]: Only "en" needs to be changed. The other language files are generated every week from Crowdin.

File: lib/OA/Admin/PasswordRecovery.php (hunk 2 above)
Comment (andrewatfornax) [reply lib/OA/Admin/PasswordRecovery.php:243]: I think this hack was supposed to be there to get the admin details of the system, to be used as who the email should be sent from.

However, the loading of the $aPref array simply didn't work at all - the password recovery system runs while there's no user logged in, so there were no user preferences available in that array.

I replaced this with the details of the system administrator name and email that can be set in the installation's configuration file, as this is definitely available when there's no one logged in, and makes the most sense to me for using in this context.

File: lib/max/language/de/default.lang.php (hunk 3 above)
Comment (andrewatfornax) [reply lib/max/language/de/default.lang.php]: Resolved.

File: lib/OA/Admin/PasswordRecovery.php (hunk 1 above)
Comment (andrewatfornax) [reply lib/OA/Admin/PasswordRecovery.php]: Yes! Done.

**Comment by andrewatfornax** [comment]
Ready for review again please, @mbeccati.

**Comment by mbeccati** [comment]
Thanks @andrewatfornax, I think we can merge!

**Pull request merged**